bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: update cluster builder and machinepools logic to support GCP

## Description
Update cluster builder and machinepools logic to support GCP.
Related to https://issues.redhat.com/browse/MGDSTRM-9129, https://issues.redhat.com/browse/MGDSTRM-9374.

Changes:
* Update the Cluster Builder logic to be able to provide GCP credentials when GCP is used
* Add `DefaultComputeMachineType` method in the Data Plane Cluster config type that returns the default compute instance type depending on the provided cloud provider

## Verification Steps

1. Modify the cloud providers configuration to support GCP. Additionally limit the kafka intsances for all instance types in all cloud providers to 0 except from the developer instances in GCP to prevent automatic creation of clusters:
```
supported_providers:
  - name: aws # name of the cloud provider
    default: true # only one default cloud provider is allowed
    regions:
      - name: us-east-1 # name of the region
        default: true # only one default region is allowed per cloud provider
        supported_instance_type:
          standard:
            limit: 0
          developer:
            limit: 0
  - name: gcp # name of the cloud provider
    default: false # only one default cloud provider is allowed
    regions:
      - name: us-east1 # name of the region
        default: true # only one default region is allowed per cloud provider
        supported_instance_type:
          standard:
            limit: 0
          developer: {}
```
2. Run kas-fleet-manager serve with the flag `--dataplane-cluster-scaling-type=auto`
3. Verify a new GCP cluster for `developer` instances is being created in GCP in region us-east1.
4. Before the cluster becomes ready modify the cloud provider config to also limit developer instances in gcp to 0. Then re-run the fleet manager with the same flags as before
5. Wait until the cluster becomes `ready` in kas fleet manager. Once it is ready verify also that addons are in ready state.



## Checklist (Definition of Done)
